### DISCUSSION TITLE
Unable to get noVNC working

### DISCUSSION THREAD

**Discussion by mthemis-provenir** [coder/coder#14034] [General]
Has anyone managed to get noVNC to work with Coder recently? I've looked at pretty much all of the examples out there, and I've tried everything, but it simply will not connect. This is the app config I'm using:

```
resource "coder_app" "novnc" {
  agent_id      = coder_agent.main.id
  slug          = "novnc"
  display_name  = "noVNC"
  icon          = "https://upload.wikimedia.org/wikipedia/commons/0/0f/NoVNC_logo.png"
  url           = "http://localhost:6901"
  subdomain     = false
  # external      = true

  healthcheck {
    url       = "http://localhost:6901"
    interval  = 5
    threshold = 6
  }
}
```

It's definitely running and listening in the pod:
```
netstat -ontpl | grep 6901
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 0.0.0.0:6901            0.0.0.0:*               LISTEN      154/python3          off (0.00/0/0)
```

If I exec into the pod and use curl, it shows the `vnc.html` page, but when using the link in the workspace, it seems to give a 404 (if I change the link to `http://localhost:6901/vnc.html`), or if not it simply fails to connect. Is there some trick I'm missing here?

**Comment by phorcys420** [comment]
Hello @mthemis-provenir, could you send over your full template alongside with the docker image (or equivalent if you're using another runtime) ?

This is pretty common with noVNC depending on how you've installed it.

Could you also send over the `curl` command that you used for testing?

**Reply by mthemis-provenir** [reply]
Thanks for getting back to me, honestly I completely forgot about this as I was trying so many things to get it going. Ultimately I managed to get it working _somehow_. I eventually implemented the wildcard setting and it seemed to work after that. I _suspect_ it's something to do with websockets, but I can't explain why the code-server is working, which apparently does use websockets, but noVNC wasn't happy when not using subdomains.

I will close this issue.
